plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/launcher/DefaultedPipelineOptionsComponent.java

```diff
@@ -131,36 +142,41 @@ public boolean isUseDefaultOptions() {
    */
   private void updateDefaultableInputValues() {
     if (isUseDefaultOptions()) {
-      customValues.put(
-          DataflowPreferences.ACCOUNT_EMAIL_PROPERTY, defaultOptions.getAccountEmail());
-      customValues.put(DataflowPreferences.PROJECT_PROPERTY, defaultOptions.getProject().getId());
-      customValues.put(
-          DataflowPreferences.STAGING_LOCATION_PROPERTY, defaultOptions.getStagingLocation());
-      // TODO: Give this a separate input
-      customValues.put(
-          DataflowPreferences.GCP_TEMP_LOCATION_PROPERTY, defaultOptions.getStagingLocation());
-      String defaultAccountEmail = preferences.getDefaultAccountEmail();
-      defaultOptions.selectAccount(Strings.nullToEmpty(defaultAccountEmail));
-      String defaultProject = preferences.getDefaultProject();
-      defaultOptions.setCloudProjectText(Strings.nullToEmpty(defaultProject));
-      String defaultStagingLocation = preferences.getDefaultStagingLocation();
-      defaultOptions.setStagingLocationText(Strings.nullToEmpty(defaultStagingLocation));
+      saveCustomValues();
```

Comment:
I think I just learned that this logic (the original logic) is weird, does not make much sense, is useless, but luckily OK because correct custom values are always loaded to the UI by `setCustomValues()`. That is, if you delete the method `saveCustomValues()`, I believe it will continue to work fine. What I think is wrong with the logic is that, if the _use defaults_ button is checked, it means the UI (`defaultOptions`) is currently showing the default (preferences) values, so it will save these preferences values to `customValues`.

However, let's not bother with it here. It's the problem of the original logic.

Reply:
Ah, I think you're seeing odd behaviour due to the call to `updateDefaultableInputValues()` in the constructor.  Looking back at the code, I added this as I was trying to mock the `RunOptionsDefaultsComponent` and I wasn't configuring it to use the defaults from the preferences unlike the real `RunOptionsDefaultsComponent`.  I'll fix that up.

Reply:
I was talking about the nonsensical behavior (I think) that is irrelevant of this PR. `customValues` is to store values entered when the _use defaults_ button is unchecked, isn't it? I was saying the code is storing values of the (project or workspace) defaults into `customValues`. That is not right, isn't it?